lib/lrmd/lrmd_client.c

```diff
@@ -1450,12 +1473,22 @@ lrmd_tcp_connect_cb(void *userdata, int rc, int sock)
         return;
     }
 
-    if (tls_client_handshake(lrmd) != pcmk_rc_ok) {
+    /* Try to do the TLS handshake once with no retry or timeouts.  If it immediately
+     * returns a result, we can go ahead and report that.  Otherwise, we have to add
+     * a trigger to the mainloop to handle retrying so that we don't block.
+     */
+    rc = pcmk__tls_client_try_handshake(native->remote, NULL);
+    if (rc == EAGAIN) {
+        native->tls_handshake_trigger = mainloop_add_trigger(G_PRIORITY_LOW, try_handshake_trigger,
+                                                             lrmd);
```

Comment:
You want to create this only if it doesn't already exist. The same trigger object can be reused any number of times